public/index.php

```diff
@@ -0,0 +1,117 @@
+    array_push($data["callIds"], $callIdString);
+  } elseif($method == 'remove') {
+    foreach ($data['callIds'] as $key => $value){
+      foreach ($value as $dKey => $dValue){
+        if($call_id == $dValue)
+          array_splice($data['callIds'], $key, 1);
+      }
+    }
+  }
+  $newJsonString = $data;
+  // var_dump($newJsonString);
+  file_put_contents($file, json_encode($newJsonString));
+  return json_encode($newJsonString);
+  // return $data;
+}
+
+
+$call_id_file = __DIR__ . '/../callId.json';
+
+// Instantiate Bandwidth Client
+$client = new BandwidthLib\BandwidthClient($config);
+$voice_client = $client->getVoice()->getClient();
+
+// Instantiate App
+$app = AppFactory::create();
+
+// Add error middleware
+$app->addErrorMiddleware(true, true, true);
+
+$app->post('/callbacks/inbound', function (Request $request, Response $response) use ($call_id_file) {
+  $data = $request->getParsedBody();
+  if ($data['eventType'] == "initiate"){
+    $arr = modifyArray($call_id_file, $data["callId"], 'add');;
+  }
+  $bxmlResponse = new BandwidthLib\Voice\Bxml\Response();
+  if ($data['eventType'] == "initiate" or $data['eventType'] == "redirect"){
+    $ring = new BandwidthLib\Voice\Bxml\Ring();
+    $ring->duration(10);
+    $redirect = new BandwidthLib\Voice\Bxml\Redirect();
+    $redirect->redirectUrl("/callbacks/inbound");
+    $bxmlResponse->addVerb($ring);
+    $bxmlResponse->addVerb($redirect);
+  }
+  $response = $response->withStatus(200)->withHeader('Content-Type', 'application/xml');
+  $response->getBody()->write($bxmlResponse->toBxml());
+  return $response;
+});
+
+
+$app->post('/callbacks/goodbye', function (Request $request, Response $response) {
+    $data = $request->getParsedBody();
+    $bxmlResponse = new BandwidthLib\Voice\Bxml\Response();
+    if ($data['eventType'] == "redirect"){
+      $speakSentence = new BandwidthLib\Voice\Bxml\SpeakSentence("The call has been updated. Goodbye");
+      $bxmlResponse->addVerb($speakSentence);
+    }
+    $response = $response->withStatus(200)->withHeader('Content-Type', 'application/xml');
+    $response->getBody()->write($bxmlResponse->toBxml());
+    return $response;
+});
+
+
+$app->delete('/calls/{id}', function (Request $request, Response $response, $args) use ($voice_client, $BANDWIDTH_ACCOUNT_ID, $BASE_URL, $call_id_file){
+    try {
+      $body = new BandwidthLib\Voice\Models\ApiModifyCallRequest();
+      $body->state = "active";
+      $body->redirectUrl = $BASE_URL."/callbacks/goodbye";
+      $voice_client->modifyCall($BANDWIDTH_ACCOUNT_ID, $args['id'], $body);
+      $arr = modifyArray($call_id_file, $args['id'], 'remove');
+      $response = $response->withStatus(200)->withHeader('Content-Type', 'application/xml');
+      $response->getBody()->write($arr);
+      return $response;
+    } catch (BandwidthLib\APIException $e) {
+      $response = $response->withStatus(404);
+      $response->getBody()->write($e);
+      return $response;
+    }
+});
```

Comment:
You could maybe clean this up by checking the existence of the ID before doing an API request. That way you can return a 404 on a call not found before making an API request, then do the API request and handle the error separately

Reply:
good idea - done